docs/graph-homomorphism-as-natural-transformation.md

```diff
@@ -5,10 +5,14 @@
  * Given a functor G from C to D.
  
 There is a natural transformation from F to G if, for every object X in category C, there is an endomorphism from F(X) to G(X).
+[Paul: Two points:
+1. The word "endomorphism" usually means a morphism from an object to itself.  In this case F(X) and G(X) are generally distinct, so you would just have a "morphism" from F(X) to G(X) rather than an "endomorphism".
```

Comment:
It sounds like you were using the word "endomorphism" the way the category theorists use the word "morphisms" - though you are correct that words like "homomorphism" and "isomorphism" predate category theory itself (I think).  Some authors use the word "arrow" instead, e.g. "let s and t be arrows from the object a to the object b".  The main point is that the morphisms are as much part of the definition of a category as the objects - indeed, it is possible (and useful!) to have two very different categories whose objects are the same but whose morphisms are different.

The word "endomorphism" is reserved specifically for a morphism from an object to itself.  Since we're already talking about the category of graphs, let's sort out the terminology in that context.  Recall that this is the category whose objects are graphs and whose morphisms are 

Let G be the graph with vertices 1, 2 and edges e(1,2) and f(2,1).  Let H be the graph with vertices 3, 4, 5 and edges g(3,4), h(4,3), i(4,5), j(5,4).  The map T: G -> H given by T(1) = 4, T(2) = 5, T(e) = h, and T(f) = i is a morphism in the category of graphs: it is not hard to check that it is a graph homomorphism, and the morphisms in the category of graphs are by definition precisely the graph homomorphisms.  But T is not an endomorphism because G and H are not the same graph.

Now consider U: G -> G defined by U(1) = 2, U(2) = 1, U(e) = f, and U(f) = e.  This is again a morphism - it satisfies the axioms of a graph homomorphism - and it is also an endomorphism since it is a morphism from G to itself.  Note that U is an isomorphism (it is its own inverse); the word for an endomorphism which is also an isomorphism is "automorphism".

Finally, consider the map W: H -> H defined by W(3) = 3, W(4) = 4, W(5) = 3, W(g) = g, W(h) = h, W(i) = h, and W(j) = g.  This is an example of an endomorphism which is not an automorphism.